### ISSUE TITLE
About performance of DGCL in addressing data sparsity

### ISSUE THREAD

**Issue by future97** [wentao228/DGCL#5]
在performance of DGCL in addressing data sparsity章节中，关于'we focus on drugs with less than 20 interactions. '的理解和实现
1） 指的是train_data加上test_data的总相互作用数据里，相互作用少于20的drug吗；还是说单纯test_data里，相互作用少于20的drug呢？
2）具体实验时，train data和test data依然是目前这样划分，但只计算test data里与这些drug相关的关联的预测性能吗

期待您的回复，非常感谢！！！

**Comment by wentao228** [comment]
您好，感谢您对我们工作的关注。
（1）这里指的是test_data里相互作用少于20的drug，因为test_data不参与构图。
（2）正如您所说，是的。

**Comment by wentao228** [comment]
您好，感谢您对我们工作的关注。
（1）这里指的是train_data里相互作用少于20的drug，因为test_data不参与构图。
（2）正如您所说，是的。

**Comment by future97** [comment]
您好，在'Visualization analysis'章节中，
1） 对相同关系样本的可视化，是通过直接拼接drug和gene的embedding（模型训练后得到的表示）来输入给t-SNE，进行展示不同样本对的表示吗？
2）文中展示的是test data上所有样本对的表示吗？

期待您的回复，万分感谢！！！

**Comment by wentao228** [comment]
您好，最好使用整合和处理后的drug和gene的embedding进行可视化，比如分类器中间层的输出。

**Comment by future97** [comment]
也即是说要是分类器MLP是两层（一层获得hidden embeddings，一层获得预测分数）的话，那么可以使用hidden embeddings来进行可视化分析是吗

**Comment by wentao228** [comment]
是的

**Comment by future97** [comment]
好的，感谢。
1）那文中t-SNE展示的是test data上所有样本对的表示吗，还是说所有样本对的表示呢
2）sensitivity的计算依据的是sklearn.metrics里的recall_score(labels, pre, average='macro')计算方法吗

期待您的回复，万分感谢！！

**Comment by future97** [comment]
sensitivity的计算依据的是sklearn.metrics里的recall_score(labels, pre, average='macro')计算方法吗

**Comment by wentao228** [comment]
（1）测试集
（2）直接用这个函数算DGCL的没问题，算CoSMIG的会出问题，最好自己用公式算
